### ISSUE TITLE
BUG: using table_conversion="matplotlib" in a script without a matplotlib figure throws an error

### ISSUE THREAD

**Issue by discdiver** [dexplo/jupyter_to_medium#26]
My script had ` table_conversion="matplotlib"` as an argument to jtm.publish().

After commented out that line, it ran fine. I'm guessing it just needs an if statement to see if there are any matplotlib  figures.

Error:

```
Traceback (most recent call last):
  File "jtm_script.py", line 3, in <module>
    jtm.publish(
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_publish_to_medium.py", line 300, in publish
    p.main()
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_publish_to_medium.py", line 218, in main
    self.md, self.image_data_dict = self.create_markdown()
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_publish_to_medium.py", line 118, in create_markdown
    mp.preprocess(self.nb, self.resources)
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/nbconvert/preprocessors/base.py", line 69, in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_preprocesors.py", line 119, in preprocess_cell
    cell['source'] = replace_md_tables(image_data_dict, cell['source'], 
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_preprocesors.py", line 55, in replace_md_tables
    md_source = table.sub(md_table_to_image, md_source)
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_preprocesors.py", line 48, in md_table_to_image
    image_data = base64.b64decode(converter(html))
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_matplotlib_table.py", line 247, in run
    self.rows, self.num_header_rows = self.parse_html(html)
  File "/Users/jeffhale/miniconda3/lib/python3.8/site-packages/jupyter_to_medium/_matplotlib_table.py", line 52, in parse_html
    val = row[j]
IndexError: list index out of range
```

**Comment by mjam03** [comment]
Hi Jeff,

I submitted an (unrelated) issue/PR on this today, but now I'm familiar with the code thought I could help out. I've tried to replicate the issue but can't manage it. I have:
 - created a dummy notebook with both a code cell and markdown cell (but no figures, images, tables):

`import pandas as pd`
This is a markdown cell

 - The following script appears to run no problem:

```
# import the publish class that does all the heavy lifting
from jupyter_to_medium._publish_to_medium import Publish

# instantiate it so the init runs and gens the useful props we want
p = Publish(filename='path_to_notebook', title='dummy', table_conversion='matplotlib', integration_token = None, pub_name = None, tags = [], publish_status='draft', notify_followers=False, license='all-rights-reserved', canonical_url=None, chrome_path=None, save_markdown=False)

# try calling the create_markdown method on it which calls the preprocessor functions
# MarkdownPreprocessor here does the work to search for imgs and tables
p.create_markdown()
```

Can you maybe share a bit more info on your specific situation then I can fix it? I think this is a great library for notebook --> medium so keen to try to improve as much as possible.

**Comment by mjam03** [comment]
I'm going to close this as I'm struggling to replicate the issue both when running from a jupyter notebook and also when using the command line interface. The one thing I have picked up is that the jupyter_to_medium package doesn't explicitly depend on pandas even though it's imported in the `_matplotlib_table.py` file where matplotlib takes screenshots. I've removed that but other than that I can't find anything wrong here.

I've also added a test now to test creation with matplotlib on the above mentioned dummy notebook (one line of code, one line of markdown, no tables).
